### PR TITLE
fix domain-server update metaverse bug

### PR DESCRIPTION
**Pull request by sethalves**
- avoid a race that can cause metaverse to not have correct ice-server information about a domain

### CODE REVIEW

**Comment by sethalves** [comment]
https://highfidelity.fogbugz.com/f/cases/1844/dev-welcome-domain-sometimes-loses-track-of-ice-server-and-the-placename-cannot-be-reached

**Comment by sethalves** [comment]
to test:
- run the servers from this PR
- set domain-server to use full automatic networking
- connect to the domain
- restart the domain-server
- connect to the domain

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9712301172a515c196a0eca9c6e0098fdf4c0160 available at: https://gist.github.com/dfcf25595ce1a55003a0f745ca89bf27<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8769-9712301.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8769-9712301.dmg

File: domain-server/src/DomainServer.cpp [hunk 1]
```diff
@@ -1317,7 +1326,19 @@ void DomainServer::sendICEServerAddressToMetaverseAPI() {
...
+        _sendICEServerAddressToMetaverseAPIRedo = false;
+        sendICEServerAddressToMetaverseAPI();
+    } else {
+        qDebug() << "ice-server address updated with metaverse.";
+    }
+}
+
 void DomainServer::handleFailedICEServerAddressUpdate(QNetworkReply& requestReply) {
+    _sendICEServerAddressToMetaverseAPIInProgress = false;
     const int ICE_SERVER_UPDATE_RETRY_MS = 2 * 1000;
```
Comment (sethalves) [inline domain-server/src/DomainServer.cpp]: If we just had a failed conversation with metaverse, shouldn't we wait a bit, either way?

Comment (birarda) [inline domain-server/src/DomainServer.cpp]: Not if we know we have a newer update that is different, IMO.

Comment (sethalves) [inline domain-server/src/DomainServer.cpp]: okay, we retry quickly if ice-server ip was updated during the previous attempt to talk to metaverse.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9712301172a515c196a0eca9c6e0098fdf4c0160 available at: https://gist.github.com/dfcf25595ce1a55003a0f745ca89bf27<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8769-9712301.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8769-9712301.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e9961110e92dba4c1b217aafed5a9688c9351954 available at: https://gist.github.com/dfcf25595ce1a55003a0f745ca89bf27<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8769-e996111.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8769-e996111.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e9961110e92dba4c1b217aafed5a9688c9351954 available at: https://gist.github.com/dfcf25595ce1a55003a0f745ca89bf27<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8769-e996111.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8769-e996111.dmg

**Comment by howard-stearns** [comment]
tested

**Pull request merged**